Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationController.java

```diff
@@ -0,0 +1,113 @@
+
+import edu.asu.spring.quadriga.domain.network.tranform.ITransformation;
+import edu.asu.spring.quadriga.exceptions.FileStorageException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaGeneratorException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.transformation.ITransformationManager;
+
+@Controller
+@Scope("session")
+public class TransformationController {
+
+    private String result;
+
+    private Future<String> future;
+
+    @Autowired
+    private ITransformationManager transformationManager;
+
+    @Autowired
+    private TransformationProcessor transformProcessor;
+
+    /**
+     * Gets transformation-request,networks from the User and returns
+     * transformed graph Extensions: Obtain Mapping and Pattern files from the
+     * user.
+     * 
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws FileStorageException
+     */
+
+    @RequestMapping(value = "auth/transformation/start", method = RequestMethod.POST)
+    public String generateTransform(@RequestParam("project") String project,
+            @RequestParam("transformation") String transformationIds,
+            @RequestParam("sanitized_network_id") String networkIds, Model model)
+            throws QuadrigaStorageException, IOException, QuadrigaGeneratorException, FileStorageException {
+
+        String[] transformationIdsList = transformationIds.split(",");
+        List<ITransformation> transformations = transformationManager.getTransformations(transformationIdsList);
+        List<String> networkIdList = Arrays.asList(networkIds.split("\\s*,\\s*"));
+
+        future = transformProcessor.runTransformationAsync(transformations, networkIdList);
+
+        return "status_transform";
+    }
+
+    /**
+     * Checks the status of future evenTt, if it's done directs to result page
+     * and if it's not done directs to status page
+     * 
+     * @return
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @RequestMapping(value = "auth/transformation/result/status", method = RequestMethod.GET)
+    public String showStatus() throws InterruptedException, ExecutionException {
+
+        if (future.isDone()) {
+            result = future.get();
+
+            if (result == null)
+                return "failure_transform";
+
+            return "success_transform";
+        }
+        return "status_transform";
+    }
+
+    /**
+     * response is built with generated graphstring and downloaded as a file
+     * 
+     * @param response
+     * @throws IOException
+     * @throws ExecutionException
+     * @throws InterruptedException
+     */
+    @RequestMapping(value = "auth/transformation/result/download")
+    public String downloadTransformation(HttpServletResponse response) throws IOException {
+
+        if (result == null)
+            return "failure_transform";
```

Comment:
If result is null, doesn't mean necessarily that there was a failure. Just means there is nothing to download.